db_checker.py

```diff
@@ -114,24 +116,20 @@ def allowed_unit(self, unit):
         """
         This method checks that the given unit conforms to standard
         """
-        if unit in allowed_units:
-            return True
+        # expand macro $(A) to a valid unit, expand $(A=B) to B
+        u = re.sub(r'\$[({].*?=(.*)?[})]', r'\1', unit)
+        u = re.sub(r'\$[({].*?[})]', 'm', u)
 
-        # otherwise check for macro
-        if "$" in unit:
-            return True
+        # split unit amalgamations
+        units = re.split(r'[/ ()]', u)
 
-        # otherwise split unit amalgamations
-        units = re.split(r'[/ ()]', unit)
         for u in units:
+            # remove any powers of units e.g. mm^2 -> mm
+            u = re.sub(r'\^[-]?\d', '', u)
```

Comment:
I can add a count to sub to make it do only one replace, then ^2^3 should fail. I think I can add a [A-Za-z]+ to the satisfy the other matter. In the future, maybe we should be using https://pypi.python.org/pypi/cf_units ?